examples/installation_on_windows/model_folder_checker.py

```diff
@@ -0,0 +1,34 @@
+import os
+def is_folder_accessible(folder_path):
+    """
+    Check if the folder exists and if it's accessible (read, write, execute).
+
+    :param folder_path: The path to the folder.
+    :return: True if the folder is accessible, False otherwise.
+    """
+    # Check if the path exists and is a directory
+    if not os.path.isdir(folder_path):
+        return False
+
+    # Check for read, write, and execute permissions
+    can_read = os.access(folder_path, os.R_OK)
+    can_write = os.access(folder_path, os.W_OK)
+    can_execute = os.access(folder_path, os.X_OK)
+
+    # The folder is accessible if all permissions are granted
+    return can_read and can_write and can_execute
+
+def main():
+
+    home_dir = os.path.expanduser('~')
+    model_path = home_dir + '/models'
+    if is_folder_accessible(model_path):
+        print('The \'~/models\' folder exists')
+    else:
+        print('Access issue detected, trying to create the model/ folder in home directory.')
+        input("Press Enter to continue...")
+        os.makedirs(model_path)
+        print('The \'~/models\' folder is successfully created')
+
+if __name__ == '__main__':
+    main()
```

Comment:
Hi, we don’t need an additional script for checking and creating folders. Our system already includes a class called `VllmModelDownloader` located in `sllm/serve/model_downloader.py`. The issue seems to be due to the default file permissions set by `shutil.copytree` and `shutil.copy`, which may limit access to the copied folder under `dest_path` in WSL. Please resolve this issue within that file, ensuring the solution doesn’t impact functionality on standard Linux systems.

Reply:
Hi, I was looking into this one today and tested the `shutil` functions that work fine. So I was trying to reproduce the model issue but didn't encounter the access issue again. Everything is downloaded and run smoothly from `pip install`. I was wondering if you updated anything related to this on your side. I'm unsure how this issue was triggered in the first place. It might be caused by something in my machine, so I updated the .md file ignoring this issue for now. But if anyone run into the same issue or if you want me to check something else specifically, I'm more than happy to do that.

Reply:
Okay.